# flatoyster_duplications
Code used for project ***k*-mer spectra and allelic coverage analyses reveal pervasive polymorphic duplications in Ostrea edulis** (Colston-Nepali, Lap√®gue and Bierne) submitted to Peer Community Journal June 2025.

### Contents ###
Download publicly available datasets: [*sra_downloads.md*](https://github.com/lilymaya/flatoyster_duplications/blob/main/sra_downloads.md)

Use *k*-mer based analyses: [*fastk_genomescope_smudgeplot.md*](https://github.com/lilymaya/flatoyster_duplications/blob/main/fastk_genomescope_smudgeplot.md)

Trim raw sequence files and map to reference genome: [*trim_and_map.md*](https://github.com/lilymaya/flatoyster_duplications/blob/main/trim_and_map.md)

Call SNPs and filter: [*call_snps_and_filter.md*](https://github.com/lilymaya/flatoyster_duplications/blob/main/call_snps_and_filter.md)

Custom python script to extract allele counts: [*create_csv.py*](https://github.com/lilymaya/flatoyster_duplications/blob/main/create_csv.py)

R script to define dosage inferred clusters: [*dosageclusters.R*](https://github.com/lilymaya/flatoyster_duplications/blob/main/dosageclusters.R)

Use CMplot to visualize density of SNPs: [*snp_density.R*](https://github.com/lilymaya/flatoyster_duplications/blob/main/snp_density.R)

Use SnpEff to annotate vcfs: [*snpeff.md*](https://github.com/lilymaya/flatoyster_duplications/blob/main/snpeff.md)

